src/Either.ts

```diff
@@ -281,7 +281,11 @@ export const tryCatch = <A>(f: Lazy<A>): Either<Error, A> => {
   try {
     return right(f())
   } catch (e) {
-    return left(e)
+    if (e instanceof Error) {
+      return left(e)
+    } else {
+      return left(new Error(String(e)))
```

Comment:
What happens if the error thrown is an object, which is so often the case is badly written JS libs? `String({ message: 'foo' }) // => [object Object]` 😒

Reply:
This exactly what I was pointing to in the issue #323

Reply:
Anyway, this bug is about incorrect return type Error for things that actually aren't Errors in tryCatch. We still need some proper solution to proccess incorrect thrown entities.